src/EnergyPlus/PluginManager.cc

```diff
@@ -1226,38 +1271,55 @@ namespace PluginManagement {
                 std::string functionName = (*EP_PyUnicode_AsUTF8)(item);
                 if (functionName == this->sHookBeginNewEnvironment) {
                     this->bHasBeginNewEnvironment = true;
+                    this->pBeginNewEnvironment = (*EP_PyString_FromString)(functionName.c_str());
```

Comment:
Store `PyObject` versions of the function names just once to avoid string operations later.